lib/controller/StudyRequestController.js

```diff
@@ -231,4 +236,214 @@ StudyRequestController.push({
   },
 });
 
+// STUDY REQUEST COMMENTS
+
+/**
+ * Create a new comment on the given study request.
+ *
+ * The request body should contain the comment in JSON format, e.g.
+ * `{ "comment": "best comment ever~~1" }`.
+ *
+ * @memberof StudyRequestController
+ * @name postStudyRequestComment
+ * @type {Hapi.ServerRoute}
+ */
+StudyRequestController.push({
+  method: 'POST',
+  path: '/requests/study/{studyRequestId}/comments',
+  options: {
+    response: {
+      schema: StudyRequestComment.read,
+    },
+    validate: {
+      params: {
+        studyRequestId: Joi.number().integer().positive().required(),
+      },
+      payload: StudyRequestComment.create,
+      query: {
+        // TODO: remove when we have RBAC
+        isSupervisor: Joi.boolean().default(false),
+      },
+    },
+  },
+  handler: async (request) => {
+    const { subject } = request.auth.credentials;
+    const { studyRequestId } = request.params;
+    const { isSupervisor } = request.query;
+    const studyRequest = await StudyRequestDAO.byId(studyRequestId);
+    if (studyRequest === null) {
+      return Boom.notFound(`no study request found with ID ${studyRequestId}`);
+    }
+    if (studyRequest.userSubject !== subject && !isSupervisor) {
+      return Boom.forbidden('cannot comment on study request owned by another user');
```

Comment:
IMHO this is a sign that we should rethink the permissions / ownership model around study requests, studies, and such.

it probably makes sense for *all* users to be able to view and comment on study requests.  this allows for out-of-reporting-chain conversations (e.g. "hey, just so you're aware, we're requesting this study over here") and allows users to understand what other requests were made previously at a given location.

so here's a possible breakdown:

- study requests:
  - create: all staff;
  - read: all staff;
  - update: only supervisors or the owner, but:
    - only supervisors can approve, assign, change status;
  - delete: not possible
- studies:
  - create: only supervisors or the owner of the study request;
  - read: all staff;
  - update: only supervisors or the owner of the study request;
  - delete: only supervisors or the owner of the study request;
- comments:
  - create: anyone;
  - read: anyone;
  - update: only the author;
  - delete: only the author.

one issue here is that these are _authorization_ decisions, which are handled at the API layer; right now, however, we test only at the DAO layer.  to test this logic, we'll need to change that.